### PR TITLE
Improve actor builders

### PR DESCRIPTION
**Pull request by didier-wenzek**
## Proposed changes

- Add a connect_consumer method to the ServiceProvider trait
- Simplify ConfigManagerBuilder::new()
- Simplify LogManagerBuilder::new()
- Simplify FirmwareManagerBuilder::new()
- Simplify LogManagerBuilder::new()
- Simplify SignalActor::builder()
- Simplify use of HealthMonitorBuilder::new()

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/extensions/tedge_health_ext/src/lib.rs [hunk 1]
```diff
@@ -20,26 +21,38 @@ use tedge_mqtt_ext::TopicFilter;
...
+        let builder = HealthMonitorBuilder {
             service_name: service_name.to_owned(),
-            subscriptions,
-            box_builder: SimpleMessageBoxBuilder::new(service_name, 16),
-        }
+            box_builder,
+        };
+
+        // Update the MQTT config
+        *mqtt.as_mut() = builder.set_init_and_last_will(mqtt.as_mut().clone());
```
Comment (albinsuresh) [inline crates/extensions/tedge_health_ext/src/lib.rs:50]: Although I'm still a bit uncomfortable with this line, and wish that there was a better MQTT config API for an actor to set this config aspect, I'm still glad that this tricky code is now limited to just this function and the usage of the `HealthMonitorBuilder` is minimal and looks very natural from the `main` functions of all the daemons.

File: crates/extensions/tedge_timer_ext/src/builder.rs [hunk 2]
```diff
@@ -49,50 +47,19 @@ impl RuntimeRequestSink for TimerActorBuilder {
 }
 
 impl<T: Message> ServiceProvider<SetTimeout<T>, Timeout<T>, NoConfig> for TimerActorBuilder {
-    fn add_peer(&mut self, peer: &mut impl ServiceConsumer<SetTimeout<T>, Timeout<T>, NoConfig>) {
-        let mut adapter = AnyTimerAdapter::new(peer);
-        self.box_builder.add_peer(&mut adapter);
-    }
-}
-
-/// A message adapter used by actors to send timer requests with a generic payload `SetTimeout<T>`
```
Comment (albinsuresh) [inline crates/extensions/tedge_timer_ext/src/builder.rs:58]: To be honest, I didn't really understand how the new `connect_consumer` API enabled this simplification. I'm struggling to understand that connection.

**Review by albinsuresh — APPROVED** [review]

File: crates/extensions/tedge_timer_ext/src/builder.rs (hunk 2 above)
Comment (didier-wenzek) [reply crates/extensions/tedge_timer_ext/src/builder.rs:58]: The former API was forcing the introduction of some concrete type implementing the `ServiceConsumer` trait. Now you can connect a `ServiceProducer` without such an implementation.

**Review by PradeepKiruvale — APPROVED** [review]
LGTM

**Pull request merged**